jbmc/src/java_bytecode/README.md

```diff
@@ -736,3 +736,151 @@ See also \ref string-solver-interface.
+`symex_dynamic::dynamic_object1` represents `globalReferenceTypeReferenceField` and `symex_dynamic::dynamic_object2` represents `referenceField`.
+- \ref symbol_exprt `symex_dynamic::dynamic_object1` = \ref struct_exprt with `referenceField` set to `null` value
+- \ref symbol_exprt `symex_dynamic::dynamic_object2` = \ref struct_exprt with `primitiveField` set to `0`
+
+Call to the constructor of `ReferenceTypePrimitiveField`, initializing the parameters `this` and `i`
+- \ref symbol_exprt `java::ReferenceTypePrimitiveField.<init>:(I)V::this` = \ref constant_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object2`
+- \ref symbol_exprt `java::ReferenceTypePrimitiveField.<init>:(I)V::i` = \ref constant_exprt with value `5`
+
+Call to the super constructor (Object). The `to_construct` parameter is internal
+- \ref symbol_exprt `java::java.lang.Object.<init>:()V::this` = \ref constant_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object2`
+- \ref symbol_exprt `java::java.lang.Object.<init>:()V::to_construct` = \ref constant_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object2`
+
+Setting the member of symex_dynamic::dynamic_object2, primitiveField. Note that primitives are always assigned using constant expressions with values.
+- \ref member_exprt `symex_dynamic::dynamic_object2.primitiveField` = \ref constant_exprt with value `5`
+
+Call to the constructor of ReferenceTypeReferenceField, initalizing the parameters `this` and `r`
+- \ref symbol_exprt `java::ReferenceTypeReferenceField.<init>:(LReferenceTypePrimitiveField;)V::this` = \ref constant_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object1`
+- \ref symbol_exprt `java::ReferenceTypeReferenceField.<init>:(LReferenceTypePrimitiveField;)V::r` = \ref constant_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object2`
+
+Call to the super constructor (Object). The `to_construct` parameter is internal
+- \ref symbol_exprt `java::java.lang.Object.<init>:()V::this` = \ref constant_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object1`
+- \ref symbol_exprt `java::java.lang.Object.<init>:()V::to_construct` = \ref constant_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object1`
+
+Setting the member of symex_dynamic::dynamic_object1, referenceField. Note that this because referenceField is of struct type, it is assigned by reference.
+- \ref member_exprt `symex_dynamic::dynamic_object1.referenceField` = \ref constant_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object2`
+
+Finally, the actual assignment to globalReferenceTypeReferenceField is done by reference to the object we built symex_dynamic::dynamic_object1
+- \ref symbol_exprt `java::ObjectReference.globalReferenceTypeReferenceField` = \ref constant_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object1`
```

Comment:
❓ OK, this section looks a little better, but what's up with these `\ref` macros?

💡 I wonder if we could shorten the names - they really clutter this example.

Reply:
I wanted to use names that would be similar to something that one would find in the trace, in case they are following along.